### PR TITLE
Fix/pandas compatibility

### PR DESCRIPTION
**Pull request by jpcampb2**
This PR addresses problems arising from misconfigured pandas version testing, to ensure proper support for stated pandas versions (see #346).

### CODE REVIEW

File: great_expectations/dataset/pandas_dataset.py [hunk 1]
```diff
@@ -192,7 +192,9 @@ def inner_wrapper(self, column, result_format = None, *args, **kwargs):
 
             element_count = int(len(series))
             nonnull_values = series[null_indexes == False]
-            nonnull_count = int((null_indexes == False).sum())
+            # Simplify this expression because the old version fails under pandas 0.21 (but only that version)
```
Comment (abegong) [inline great_expectations/dataset/pandas_dataset.py:195]: +1
Good commenting.

File: great_expectations/dataset/pandas_dataset.py [hunk 2]
```diff
@@ -1082,7 +1084,8 @@ def expect_column_chisquare_test_p_value_to_be_greater_than(self, column, partit
         # Convert to Series object to allow joining on index values
         expected_column = pd.Series(partition_object['weights'], index=partition_object['values'], name='expected') * len(column)
         # Join along the indices to allow proper comparison of both types of possible missing values
-        test_df = pd.concat([expected_column, observed_frequencies], axis=1, sort=True)
+        # test_df = pd.concat([expected_column, observed_frequencies], axis=1, sort=True) # Sort parameter not available before pandas 0.23.0
```
Comment (abegong) [inline great_expectations/dataset/pandas_dataset.py:1087]: Why did we include `sort=True` to begin with?

File: requirements-dev.txt [hunk 3]
```diff
@@ -7,7 +7,7 @@ six>=1.9.0
 jsonschema>=2.5.1
 sqlalchemy>=1.2
 xlrd>=1.1.0
-pyarrow>=0.9.0
+pyarrow==0.9.0
```
Comment (abegong) [inline requirements-dev.txt:10]: Thanks for figuring this out. Seemed like a real pain to get to the bottom of it.

**Review by abegong — APPROVED** [review]
Looks good. Thanks for chasing this down!

File: great_expectations/dataset/pandas_dataset.py (hunk 2 above)
Comment (jpcampb2) [reply great_expectations/dataset/pandas_dataset.py:1087]: No, I added it because *not* including it throws a warning in pandas 0.23.

**Pull request merged**